src/app/sign-up/sign-up.component.ts

```diff
@@ -1,15 +1,49 @@
 import { Component, OnInit } from '@angular/core';
-
+import {SharedService} from '../shared.service';
+import {CommonService} from '../common.service';
+import {NotificationsService, SimpleNotificationsModule } from 'angular2-notifications';
 @Component({
-  selector: 'app-sign-up',
-  templateUrl: './sign-up.component.html',
-  styleUrls: ['./sign-up.component.css']
+	selector: 'app-sign-up',
+	templateUrl: './sign-up.component.html',
+	styleUrls: ['./sign-up.component.css']
 })
 export class SignUpComponent implements OnInit {
+	user = {firstName:'' ,lastName:'' ,Email:'', mobileNumber:"" ,username:'' ,password:'' ,confirmPassword:''};
 
-  constructor() { }
+	constructor(public sharedService:SharedService, public notificationService:NotificationsService, public commonService:CommonService) { }
+	public data : any;
+	ngOnInit() {
+	}
+	registerUser(){
+		console.log(this.user);
+		 if (this.commonService.required(this.user.firstName) && this.commonService.required(this.user.lastName) && this.commonService.required(this.user.Email) && this.commonService.required(this.user.username) && this.commonService.required(this.user.password) && this.commonService.required(this.user.confirmPassword)) {
+		
+		 	if(this.commonService.checkValidEmail(this.user.Email)){
+				
+				if(this.commonService.checkValidMobileNumber(this.user.mobileNumber)) {
 
-  ngOnInit() {
-  }
+					if (this.commonService.matchPassword(this.user)) {
+					
+					}else{
+						alert('not match');
+					}
+					
+				}else{
+					this.notificationService.error('Error','Invalid mobilenumber');
+				}
+		 	}else{
+		 		this.notificationService.error('Error','Invalid email');
+			}
+		 }else{
+		 	this.notificationService.error('Error','Please fill all the required fields');
+		}
 
+		// this.sharedService.registerUser(this.user).subscribe(res =>{
```

Comment:
Commented code will never be pushed to live repo. Either uncomment it or delete it.